argocd/plugin.yaml

```diff
@@ -0,0 +1,112 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ConfigManagementPlugin
+metadata:
+  name: thelma
+spec:
+  # Thelma has a config profile named "argocd" that provides helpful defaults for running Thelma within an ArgoCD on GKE.
+  # This file doesn't offer a clean way to specify the THELMA_CONFIG_PROFILE environment variable to set it (at least,
+  # not in any way where it could be safely overridden later) so we leave that up whoever is configuring this plugin with
+  # ArgoCD. Since ArgoCD isn't interactive, the "ci" profile will be chosen by default.
+  #
+  # https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/#write-the-plugin-configuration-file
+  generate:
+    command: ["thelma", "render"]
+    args: ["--flags-from-environment-prefix=PARAM_"]
+  # Parameter declarations are for UI sugar only. They have no bearing on what actually gets passed to the Thelma:
+  # even the `string` default values aren't passed, and it's up to Thelma to supply these defaults internally.
+  #
+  # ArgoCD will convert parameter names to environment variables by uppercasing them and replacing dashes with
+  # underscores with a PARAM_ prefix, same as how Thelma handles --flags-from-environment-prefix=PARAM_.
+  #
+  # https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/#using-environment-variables-in-your-plugin
+  parameters:
```

Comment:
curious to see how this shows up in the ArgoCD UI!

Reply:
Honestly same, I think I'll certainly go back and update here depending on how the documentation shows up. I don't expect this to be end-user-friendly but my hope is that it'll be easier for our team to debug